lib/web_http.dart

```diff
@@ -0,0 +1,149 @@
+import 'dart:async';
+import 'dart:html';
+
+/// A web http client focused on speed and correctness.
+///
+/// Unlike other http clients, this client is based around streams to allow
+/// cancelation of extra requests.
+class HttpClient {
+  /// Creates a new [HttpClient].
+  const HttpClient();
+
+  /// A convience wrapper around [request] which sends a GET request.
+  ///
+  /// See also:
+  ///
+  ///   * [request], for a description of the method parameters.
+  Stream<String> get(
+    String url, {
+    Map<String, String> headers,
+    int timeout,
+    bool withCredentials,
+  }) {
+    return request(url, 'GET', headers: headers, timeout: timeout)
+        .map((HttpResponse response) => response.body);
+  }
+
+  /// A convience wrapper around [request] which sends a POST request.
+  ///
+  /// See also:
+  ///
+  ///   * [request], for a description of the method parameters.
+  Stream<String> post(
+    String url,
+    String body, {
+    Map<String, String> headers,
+    int timeout,
+    bool withCredentials,
+  }) {
+    return request(url, 'POST', headers: headers, timeout: timeout)
+        .map((HttpResponse response) => response.body);
+  }
+
+  /// Sends an http request to a specified domain.
+  ///
+  /// [url] is the String encoded url.
```

Comment:
nit: Your args comments aren't formatted consistently here in comparison to the previous two methods.